engine/src/build/mac/apple_sdk.py

```diff
@@ -0,0 +1,171 @@
+# Copyright 2013 The Flutter Authors. All rights reserved.
+# Use of this source code is governed by a BSD-style license that can be
+# found in the LICENSE file.
+
+import argparse
+import os
+import shutil
+import subprocess
+import sys
+
+from dataclasses import dataclass
+
+# Imports from //flutter/build/pyutil.
+sys.path.insert(1, os.path.join(os.path.dirname(__file__), os.pardir))
+from pyutil.file_util import symlink
+
+# This script creates symlinks to Apple SDKs, Platforms, and host toolchain
+# under //flutter/prebuilts.
+PREBUILTS = os.path.realpath(os.path.join(
+  os.path.dirname(__file__), os.pardir, os.pardir, 'flutter', 'prebuilts',
+))
+
+# Supported SDKs.
+SDKS = ['iphoneos', 'iphonesimulator', 'macosx']
+
+
+def parse_arguments():
```

Comment:
ask to learn: what's the relationship between this python script and the above gni file?

Reply:
This python script is called from two places:
* As a gclient hook when you `gclient sync` -- see the invocation in `DEPS`. This is used to pre-populate the symlinks under //flutter/prebuilts/SDKs and //flutter/prebuilts/Platforms.
* During a `gn` invocation _if required_ -- i.e. if we haven't passed in the appropriate variables from `//flutter/tools/gn`. Notice that the `//flutter/tools/gn` script goes looking for the symlinks that were created during gclient sync and uses them to pre-populate the required variables, which then avoids us having to run the script during each build.